### PR TITLE
[PLT-6676] Make OutgoingWebhook to fire when post has no text content but only attachment

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Make OutgoingWebhook to fire when post has no text content but only attachment

#### Ticket Link
Jira ticket: [PLT-6676](https://mattermost.atlassian.net/browse/PLT-6676)

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to the drivers
- [x] Touches critical sections of the codebase (auth, upgrade, etc.)

### CODE REVIEW

File: model/outgoing_webhook.go [hunk 1]
```diff
@@ -199,7 +201,9 @@ func (o *OutgoingWebhook) PreUpdate() {
 }
 
 func (o *OutgoingWebhook) HasTriggerWord(word string) bool {
-	if len(o.TriggerWords) == 0 || len(word) == 0 {
+	if len(o.TriggerWords) == 0 {
+		return true
```
Comment (ccbrown) [inline model/outgoing_webhook.go]: This makes the behavior of these two functions really unintuitive given the names (I wouldn't expect `OutgoingWebhook{}.HasTriggerWord("foo")` to return `true`.). It seems like your other changes should fix the issue, so is this change actually necessary?

Comment (saturninoabril) [inline model/outgoing_webhook.go]: Noted. Renamed the function to `TriggerWordExactMatch`. Also added another one as `GetTriggerWord `

**Comment by saturninoabril** [comment]
@ccbrown thanks! see updated PR.
